spec/elastic_apm/normalizers/active_record_spec.rb

```diff
@@ -53,6 +55,49 @@ def normalize_payload(payload)
             expect(context_.db.statement).to eq sql
           end
 
+          it 'uses the connection from payload, when available' do
+            allow(::ActiveRecord::Base)
+              .to receive(:connection) { double(adapter_name: nil) }
+            sql = 'SELECT  "burgers".* FROM "burgers" ' \
+              'WHERE "burgers"."cheese" = $1 LIMIT 1'
+
+            name, type, subtype, action, context_ = normalize_payload(
+              sql: sql,
+              connection: double(adapter_name: 'MySQL')
+            )
+
+            expect(name).to eq 'SELECT FROM burgers'
+            expect(type).to eq 'db'
+            expect(subtype).to eq 'mysql'
+            expect(action).to eq 'sql'
+            expect(context_.db.statement).to eq sql
+
+            name, type, subtype, action, context_ = normalize_payload(
+              sql: sql,
+              connection: double(adapter_name: 'Postgres')
+            )
+
+            expect(name).to eq 'SELECT FROM burgers'
+            expect(type).to eq 'db'
+            expect(subtype).to eq 'postgres'
+            expect(action).to eq 'sql'
+            expect(context_.db.statement).to eq sql
+
+            sql = 'SELECT  "pizza".* FROM "pizza" ' \
+              'WHERE "pizza"."topping" = $1 LIMIT 1'
+
+            name, type, subtype, action, context_ = normalize_payload(
+              sql: sql,
+              connection: double(adapter_name: nil)
+            )
+
+            expect(name).to eq 'SELECT FROM pizza'
+            expect(type).to eq 'db'
+            expect(subtype).to eq 'unknown'
+            expect(action).to eq 'sql'
+            expect(context_.db.statement).to eq sql
+          end
```

Comment:
Just my tastes, but I don't think you need to expect every value with every adapter. Keeping it to the expected changes makes it easier to read  the test and understand what's under the microscope.